src/core/public/keyboard_shortcut/keyboard_shortcut_service.ts

```diff
@@ -18,6 +13,7 @@ import { KeyboardShortcutSetup, KeyboardShortcutStart, ShortcutDefinition } from
 export class KeyboardShortcutService {
   private shortcutsMapByKey = new Map<string, ShortcutDefinition[]>();
   private namespacedIdToKeyLookup = new Map<string, string>();
+  private keyParser = new KeyStringParser();
```

Comment:
do you need this keyparser? Does keyparser contain any state of it's own? Can it just be used statically?